prog/hints.go

```diff
@@ -27,26 +27,27 @@ import (
 	"github.com/google/syzkaller/pkg/image"
 )
 
-// Example: for comparisons {(op1, op2), (op1, op3), (op1, op4), (op2, op1)}
-// this map will store the following:
-//
-//	m = {
-//			op1: {map[op2]: true, map[op3]: true, map[op4]: true},
-//			op2: {map[op1]: true}
-//	}.
-type CompMap map[uint64]map[uint64]bool
+// CompMap maps comparison operand that could come from the input to the second operand to the PC.
+type CompMap map[uint64]map[uint64]map[uint64]bool
```

Comment:
Nit: just a thought that came to my mind while re-reading this code.

Could it be easier (and even more efficient) to use `type CompMap map[Comp]bool`, with `Comp` being a struct of 3 `uint64`?

Reply:
The current algo needs to walk over all comparison with the same first operand.
And in limitAttempts we want to check all PCs for the same second operand.
And we also want to deduplicate entries.
Multi-level maps allow to easily do all of this.